src/MOI_wrapper.jl

```diff
@@ -0,0 +1,225 @@
+import MathOptInterface as MOI
+using ManifoldsBase: ManifoldsBase
+using Manifolds: Manifolds
+
+include("qp_block_data.jl")
+
+const _FUNCTIONS = Union{
+    MOI.VariableIndex,MOI.ScalarAffineFunction{Float64},MOI.ScalarQuadraticFunction{Float64}
+}
+
+struct VectorizedManifold{M} <: MOI.AbstractVectorSet
+    manifold::M
+end
+function MOI.dimension(::VectorizedManifold{<:Manifolds.Sphere{N}}) where {N}
```

Comment:
We have something similar (though probably not the same) that is `representation_size`. For those manifolds where we have a representation as a single matrix/array it returns the size of these https://juliamanifolds.github.io/ManifoldsBase.jl/stable/functions.html#ManifoldsBase.representation_size-Tuple{AbstractManifold}